client/api/analytics/AnalyticsPublisherService.cs

```diff
@@ -94,15 +90,14 @@ public void SendDataAndResetCache()
                 SetMetricsAttributes(metricsData, VariationValueAttribute, evaluation.Variation.Value);
                 SetMetricsAttributes(metricsData, TargetAttribute, evaluation.Target.Identifier);
                 SetCommonSdkAttributes(metricsData);
-                StagingSeenTargets.TryAdd(evaluation.Target, 0);
                 metrics.MetricsData.Add(metricsData);
             }
 
             // Handle TargetAnalytics
             foreach (var targetAnalytic in targetsCache)
             {
                 var target = targetAnalytic.Key.Target;
-                if (target != null && !SeenTargets.ContainsKey(target) && !target.IsPrivate)
```

Comment:
We now do the `isPrivate` check at time of pushing the target to the cache, post-evaluation. Which is much more efficient